packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -0,0 +1,434 @@
+                writable: false,
+            };
+		}
+		return undefined;
+	},
+};
+
+/**
+ * See {@link UnwrappedEditableTree} for documentation on what unwrapping this perform.
+ */
+function inProxyOrUnwrap(target: ProxyTarget): UnwrappedEditableTree {
+    if (isPrimitive(target.type)) {
+        const nodeValue = target.cursor.value;
+        // see `isPrimitive` doc for `undefined` case
+        assert(nodeValue !== undefined, "Undefined values not allowed");
+        if (isPrimitiveValue(nodeValue)) {
+            return nodeValue;
+        }
+    }
+    const primary = target.getPrimaryArrayKey();
+    if (primary !== undefined) {
+        const childTargets = mapCursorField(target.cursor, primary, (c) => new ProxyTarget(target.context, c));
+        return childTargets.map(inProxyOrUnwrap);
+    }
+    return adaptWithProxy(target, handler);
+}
+
+/**
+ * @param fieldKind - determines how return value should be typed. See {@link UnwrappedEditableField}.
+ * @param childTargets - targets for the children of the field.
+ */
+function proxifyField(fieldKind: FieldKind, childTargets: ProxyTarget[]): UnwrappedEditableField {
+    if (fieldKind.multiplicity === Multiplicity.Sequence) {
+        // Return array for sequence fields
+        return childTargets.map(inProxyOrUnwrap);
+    } else {
+        // Avoid wrapping non-sequence fields in arrays
+        assert(childTargets.length <= 1, "invalid non sequence");
+        if (childTargets.length === 1) {
+            return inProxyOrUnwrap(childTargets[0]);
+        } else {
+            return undefined;
+        }
+    }
+}
+
+/**
+ * A simple API for a Forest to showcase basic interaction scenarios.
+ *
+ * This function returns an instance of a JS Proxy typed as an EditableTree.
+ * Use built-in JS functions to get more information about the data stored e.g.
+ * ```
+ * const [context, data] = getEditableTree(forest);
+ * for (const key of Object.keys(data)) { ... }
+ * // OR
+ * if ("foo" in data) { ... }
+ * context.free();
+ * ```
+ *
+ * Not (yet) supported: create properties, set values and delete properties.
+ *
+ * @returns {@link EditableTree} for the given {@link IEditableForest}.
+ * Also returns an {@link EditableTreeContext} which is used manage the cursors and anchors within the EditableTrees:
+ * This is necessary for supporting using this tree across edits to the forest, and not leaking memory.
+ */
+export function getEditableTree(forest: IEditableForest): [EditableTreeContext, UnwrappedEditableField] {
+	const context = new ProxyContext(forest);
+    const cursor = forest.allocateCursor();
+    const destination = forest.root(forest.rootField);
+    const cursorResult = forest.tryMoveCursorTo(destination, cursor);
+    const targets: ProxyTarget[] = [];
+    if (cursorResult === TreeNavigationResult.Ok) {
+        do {
+            targets.push(new ProxyTarget(context, cursor));
+        } while (cursor.seek(1) === TreeNavigationResult.Ok);
+    }
+    cursor.clear();
```

Comment:
this is unneeded. If we wanted to reuse this cursor later, we could clear it, but since its just getting freed below this, there is no benifit, and just added overhead.
```suggestion
```

Reply:
But then it will stay in a forest forever, isn't it? Meaning also that you can't edit the forest neither clear this cursor from outside the forest API, looks like a dead end.

Reply:
Clearing allows you to edit despite the cursor not being freed (edits the cursor to remove its position). Free deletes the cursor from the forest (making its position not matter), which also allows you to edit. Since the line after this is free, clearing is unneeded.

Reply:
I have implemented some tests for edits in https://github.com/sharptrip/FluidFramework/pull/4. If this line is removed, I immediately get `Error: 0x374` from the ObjectForest.

I think it does not work as you said, at least, I don't see where `free()` deletes the cursor:

https://github.com/microsoft/FluidFramework/blob/91153dd3359cee47966e7927db8a2df8d267a980/packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts#L242-L245

Reply:
That is a bug in object forest. I'll fix it and clean up this extra clear call when I do. Good find!

Reply:
I pushed a fix for this.

Reply:
> I pushed a fix for this.

Thanks! That's basically my problem here: it's still hard to say, if there is a bug or a misuse from my side.